File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/examples-java/virtualMachineExamples/VirtualMachines_List_MinimumSet_Gen.java
import com.azure.core.util.Context;

/** Samples for VirtualMachines ListByResourceGroup. */
public final class Main {
    /*
     * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/examples/virtualMachineExamples/VirtualMachines_List_MinimumSet_Gen.json
     */
    /**
     * Sample code: VirtualMachines_List_MinimumSet_Gen.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void virtualMachinesListMinimumSetGen(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .virtualMachines()
            .manager()
            .serviceClient()
            .getVirtualMachines()
            .listByResourceGroup("rgcompute", null, Context.NONE);
    }
}

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/examples-js/virtualMachineExamples/VirtualMachines_List_MinimumSet_Gen.js
const { ComputeManagementClient } = require("@azure/arm-compute");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Lists all of the virtual machines in the specified resource group. Use the nextLink property in the response to get the next page of virtual machines.
 *
 * @summary Lists all of the virtual machines in the specified resource group. Use the nextLink property in the response to get the next page of virtual machines.
 * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-01/ComputeRP/examples/virtualMachineExamples/VirtualMachines_List_MinimumSet_Gen.json
 */
async function virtualMachinesListMinimumSetGen() {
  const subscriptionId = "{subscription-id}";
  const resourceGroupName = "rgcompute";
  const credential = new DefaultAzureCredential();
  const client = new ComputeManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.virtualMachines.list(resourceGroupName)) {
    resArray.push(item);
  }
  console.log(resArray);
}

virtualMachinesListMinimumSetGen().catch(console.error);
